// @pokertools/evaluator/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCardCodes = exports.getCardCode = exports.stringifyCardCode = exports.HAND_RANK_DESCRIPTIONS = exports.HandRank = void 0;
exports.evaluate = evaluate;
exports.evaluateStrings = evaluateStrings;
exports.evaluateBoard = evaluateBoard;
exports.rank = rank;
exports.rankBoard = rankBoard;
exports.rankDescription = rankDescription;
const evaluator_1 = require("./core/evaluator");
const card_1 = require("./utils/card");
const hand_rank_1 = require("./models/hand-rank");
// Export Constants and Enums for consumers
var hand_rank_2 = require("./models/hand-rank");
Object.defineProperty(exports, "HandRank", { enumerable: true, get: function () { return hand_rank_2.HandRank; } });
Object.defineProperty(exports, "HAND_RANK_DESCRIPTIONS", { enumerable: true, get: function () { return hand_rank_2.HAND_RANK_DESCRIPTIONS; } });
var card_2 = require("./utils/card");
Object.defineProperty(exports, "stringifyCardCode", { enumerable: true, get: function () { return card_2.stringifyCardCode; } });
Object.defineProperty(exports, "getCardCode", { enumerable: true, get: function () { return card_2.getCardCode; } });
Object.defineProperty(exports, "getCardCodes", { enumerable: true, get: function () { return card_2.getCardCodes; } });
/**
 * Primary entry point.
 * Evaluates 5, 6, or 7 card integer codes.
 * Returns a strength score (lower is better).
 *
 * @param codes Array of card integers (e.g. generated by getCardCode)
 */
function evaluate(codes) {
    const len = codes.length;
    if (len === 7)
        return (0, evaluator_1.evaluate7Cards)(codes);
    if (len === 6)
        return (0, evaluator_1.evaluate6Cards)(codes);
    if (len === 5)
        return (0, evaluator_1.evaluate5Cards)(codes);
    throw new Error(`Evaluator requires 5, 6, or 7 cards. Received ${len}.`);
}
/**
 * Evaluates an array of card strings (e.g. ['As', 'Kd', ...]).
 * Slightly slower than `evaluate` due to parsing overhead.
 */
function evaluateStrings(cards) {
    return evaluate((0, card_1.getCardCodes)(cards));
}
/**
 * Evaluates a board string (e.g. "As Kd Qh ...").
 */
function evaluateBoard(board) {
    return evaluate((0, card_1.getBoardCodes)(board));
}
/**
 * Returns the HandRank (enum 0-8) for a given set of card integers.
 * Useful for categorizing hands (e.g. "Full House").
 */
function rank(codes) {
    return (0, hand_rank_1.getHandRank)(evaluate(codes));
}
/**
 * Returns the HandRank (enum 0-8) for a given board string.
 */
function rankBoard(board) {
    return (0, hand_rank_1.getHandRank)(evaluateBoard(board));
}
/**
 * Returns the human-readable name of the hand rank (e.g. "Straight Flush").
 */
function rankDescription(rank) {
    return hand_rank_1.HAND_RANK_DESCRIPTIONS[rank];
}
