// @pokertools/evaluator/dist/utils/card.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCardCode = getCardCode;
exports.getCardCodes = getCardCodes;
exports.getBoardCodes = getBoardCodes;
exports.stringifyCardCode = stringifyCardCode;
/**
 * Maps rank characters to their integer index (2=0 ... A=12)
 */
const RANK_MAP = {
    "2": 0,
    "3": 1,
    "4": 2,
    "5": 3,
    "6": 4,
    "7": 5,
    "8": 6,
    "9": 7,
    T: 8,
    J: 9,
    Q: 10,
    K: 11,
    A: 12,
};
/**
 * Maps suit characters to their integer index (s=0, h=1, d=2, c=3)
 */
const SUIT_MAP = {
    s: 0,
    h: 1,
    d: 2,
    c: 3,
};
const RANK_CHARS = ["2", "3", "4", "5", "6", "7", "8", "9", "T", "J", "Q", "K", "A"];
const SUIT_CHARS = ["s", "h", "d", "c"];
/**
 * Converts a card string (e.g. "Ah", "Td") into the integer format
 * expected by the evaluator.
 *
 * Format: (RankIndex << 2) | SuitIndex
 */
function getCardCode(cardStr) {
    if (cardStr.length !== 2) {
        throw new Error(`Invalid card string length: "${cardStr}"`);
    }
    const r = RANK_MAP[cardStr[0]];
    const s = SUIT_MAP[cardStr[1]];
    if (r === undefined || s === undefined) {
        throw new Error(`Invalid card characters: "${cardStr}"`);
    }
    return (r << 2) | s;
}
/**
 * Converts an array of card strings into an array of integers.
 */
function getCardCodes(cards) {
    const len = cards.length;
    const out = new Array(len);
    for (let i = 0; i < len; i++) {
        out[i] = getCardCode(cards[i]);
    }
    return out;
}
/**
 * Parses a board string (e.g. "Ah Ks Qd") into integers.
 */
function getBoardCodes(board) {
    if (!board)
        return [];
    return getCardCodes(board.trim().split(/\s+/));
}
/**
 * Converts an integer code back to a string (e.g. 49 -> "Ah").
 */
function stringifyCardCode(code) {
    const rank = code >> 2;
    const suit = code & 0b11;
    return (RANK_CHARS[rank] || "?") + (SUIT_CHARS[suit] || "?");
}
