// @pokertools/evaluator/dist/models/hand-rank.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HAND_RANK_DESCRIPTIONS = exports.HandRank = void 0;
exports.getHandRank = getHandRank;
/**
 * Standard Poker Hand Ranks.
 * Using const enum for inlining performance.
 */
var HandRank;
(function (HandRank) {
    HandRank[HandRank["StraightFlush"] = 0] = "StraightFlush";
    HandRank[HandRank["FourOfAKind"] = 1] = "FourOfAKind";
    HandRank[HandRank["FullHouse"] = 2] = "FullHouse";
    HandRank[HandRank["Flush"] = 3] = "Flush";
    HandRank[HandRank["Straight"] = 4] = "Straight";
    HandRank[HandRank["ThreeOfAKind"] = 5] = "ThreeOfAKind";
    HandRank[HandRank["TwoPair"] = 6] = "TwoPair";
    HandRank[HandRank["OnePair"] = 7] = "OnePair";
    HandRank[HandRank["HighCard"] = 8] = "HighCard";
})(HandRank || (exports.HandRank = HandRank = {}));
exports.HAND_RANK_DESCRIPTIONS = {
    [0 /* HandRank.StraightFlush */]: "Straight Flush",
    [1 /* HandRank.FourOfAKind */]: "Four of a Kind",
    [2 /* HandRank.FullHouse */]: "Full House",
    [3 /* HandRank.Flush */]: "Flush",
    [4 /* HandRank.Straight */]: "Straight",
    [5 /* HandRank.ThreeOfAKind */]: "Three of a Kind",
    [6 /* HandRank.TwoPair */]: "Two Pair",
    [7 /* HandRank.OnePair */]: "One Pair",
    [8 /* HandRank.HighCard */]: "High Card",
};
/**
 * Converts a raw evaluator score into a HandRank category.
 * Thresholds are based on the specific hash algorithm used.
 */
function getHandRank(val) {
    if (val > 6185)
        return 8 /* HandRank.HighCard */; // 1277 high cards
    if (val > 3325)
        return 7 /* HandRank.OnePair */; // 2860 one pairs
    if (val > 2467)
        return 6 /* HandRank.TwoPair */; //  858 two pairs
    if (val > 1609)
        return 5 /* HandRank.ThreeOfAKind */; //  858 three-kinds
    if (val > 1599)
        return 4 /* HandRank.Straight */; //   10 straights
    if (val > 322)
        return 3 /* HandRank.Flush */; // 1277 flushes
    if (val > 166)
        return 2 /* HandRank.FullHouse */; //  156 full houses
    if (val > 10)
        return 1 /* HandRank.FourOfAKind */; //  156 four-kinds
    return 0 /* HandRank.StraightFlush */; //   10 straight-flushes
}
